test/sidecar_test.go

```diff
@@ -47,26 +47,32 @@ func TestSidecarTaskSupport(t *testing.T) {
 		stepCommand    []string
 		sidecarCommand []string
 	}{{
-		desc:           "A sidecar that runs forever is terminated when Steps complete",
-		stepCommand:    []string{"echo", "\"hello world\""},
-		sidecarCommand: []string{"sh", "-c", "while [[ true ]] ; do echo \"hello from sidecar\" ; done"},
+		desc:        "A sidecar that runs forever is terminated when Steps complete",
+		stepCommand: []string{"echo", "\"hello world\""},
+		// trapping the exit signals lets us stop the sidecar more
+		// rapidly than waiting for a force stop.
```

Comment:
one detail thats missing for me is why trapping the signal is faster - i assume it's b/c whatever the default handler is for the signal, it's slower? (i.e. the stop is still being forced regardless i assume in that the controller is trying to stop the sidecar, but its a matter of how quickly the signal is handled?)

Reply:
> My working theory is that kubernetes treats the termination of the container as a graceful shutdown, so first sends sigterm and then waits 30 seconds before sending sigkill.

It might rely on the underlying container runtime, but that is most likely what you describe — at least that's how the docker runtime (and most likely containerd too) works when you ask it to stop, it sends SIGTERM, and wait for a given time for the container to stop itself, before sending SIGKILL.